controller/workitem.go

```diff
@@ -561,6 +563,199 @@ func getVersion(version interface{}) (int, error) {
+				columnMap[csvFieldKey] = len(csvGrid[0])
+				// add the new header entry.
+				csvGrid[0] = append(csvGrid[0], csvFieldKey)
+				// append empty values to all existing "lines" for the new column.
+				for k := 1; k < len(csvGrid); k++ {
+					csvGrid[k] = append(csvGrid[k], "")
+				}
+			}
+			// now we are ready to put the value where is belongs in the grid.
+			gridRowIdx := len(csvGrid) - 1 // last line.
+			gridColumnIdx := columnMap[csvFieldKey]
+			csvGrid[gridRowIdx][gridColumnIdx] = fieldValues[j]
+		}
+	}
+	// create CSV from the [][]string
+	buf := new(bytes.Buffer)
+	w := csv.NewWriter(buf)
+	err := w.WriteAll(csvGrid)
+	if err != nil {
+		return "", errs.Wrapf(err, "failed to serialize to CSV format")
+	}
+	// done, return serialized result.
+	return buf.String(), nil
+}
+
+// ConvertWorkItemToStringValue is responsible for converting given WorkItem model object into a
+// string slice; it returns the field names, the field keys and the string converted values for the work item
+func ConvertWorkItemToStringValue(ctx context.Context, app application.Application, uuidStringCache *map[string]string, wit workitem.WorkItemType, wi workitem.WorkItem) ([]string, []string, []string, error) {
+	fieldNames := []string{}
+	fieldKeys := []string{}
+	fieldValues := []string{}
+	for fieldKey, fieldDefinition := range wit.Fields {
+		// convert the value to a string for the CSV
+		fieldNames = append(fieldNames, fieldDefinition.Label)
+		fieldKeys = append(fieldNames, fieldKey)
+		fieldValueGeneric := wi.Fields[fieldKey]
+		fieldType := fieldDefinition.Type
+		fieldValueStr, err := fieldType.ConvertToStringSlice(fieldValueGeneric)
+		if err != nil {
+			return nil, nil, nil, errs.Wrapf(err, "failed to convert simple type value to string for field key: %s", fieldKey)
+		}
+		var convertedValue *string
+		// now retrieve and, if needed, resolve the id value.
+		switch fieldType.(type) {
+		case workitem.ListType:
+			var converted string
+			kind := fieldType.(workitem.ListType).ComponentType.Kind
+			delim := ""
+			for _, elem := range fieldValueStr {
+				elemConvertedValue, err := convertValueToString(ctx, app, uuidStringCache, fieldValueGeneric, []string{elem}, fieldKey, kind)
+				if err != nil {
+					return nil, nil, nil, errs.Wrapf(err, "failed to convert compound type value to string for field key: %s", fieldKey)
+				}
+				converted = converted + delim + *elemConvertedValue
+				delim = ";"
+			}
+			convertedValue = ptr.String(converted)
+		case workitem.EnumType:
+			kind := fieldType.(workitem.EnumType).BaseType.Kind
+			convertedValue, err = convertValueToString(ctx, app, uuidStringCache, fieldValueGeneric, fieldValueStr, fieldKey, kind)
+		default:
+			// all other Kinds don't need compound resolving.
+			kind := fieldType.GetKind()
+			convertedValue, err = convertValueToString(ctx, app, uuidStringCache, fieldValueGeneric, fieldValueStr, fieldKey, kind)
+		}
+		if err != nil {
```

Comment:
This err could also be from `case workitem.EnumType` block. You might want to either 
 - move the `err != nil` inside the `case workitem.Enumtype` and `default` block
or
 - change the error message on line 673 to a more generic error. Like `failed to resolve type value to string..`

Reply:
Why? In all cases @michaelkleinhenz assigns (not instantiates) to the `err` object.

Reply:
> Why? In all cases @michaelkleinhenz assigns (not instantiates) to the `err` object.

I commented on the wrong line. I was talking about the next line which is
```
return nil, nil, nil, errs.Wrapf(err, "failed to resolve simple type value to string for field key: %s", fieldKey)
```
The error message seems to be off and that's why I made the suggestion above.

Reply:
Fixed 7ae9140. Made the error msg clearer.